### PR TITLE
gh-124471: Set name for unnamed reusable workflow

### PR DESCRIPTION
**Pull request by corona10**
* Issue: gh-124471

### CODE REVIEW

File: .github/workflows/reusable-macos.yml [hunk 1]
```diff
@@ -1,3 +1,5 @@
+name: macOS
```
Comment (rruuaanng) [inline .github/workflows/reusable-macos.yml:1]: Hmmm, To me, this name is a bit overkill since the filename already shows which platform it’s for.

**Comment by hugovk** [comment]
Please can you share a before and after screenshot of what actually changed in the UI?

**Comment by corona10** [comment]
Screenshot from 2024-09-24 20-15-02

@hugovk Can you see it how it is changed?

**Comment by hugovk** [comment]
Yes, now I see. Here is how it looks now (at https://github.com/python/cpython/actions):

image

Although I'm not sure if we want these actual names, because if you click the "macOS" one, you don't see anything useful:

image

These are reusable workflows, which are like include files for the main "Tests" build. People might be confused why they don't see the macOS builds here.

**Comment by corona10** [comment]
> Although I'm not sure if we want these actual names, because if you click the "macOS" one, you don't see anything useful:

Agree, then we have to leave as-is or define somekinds special naming that people notice that it's not a actual workflow. 
(e.g, [reusable workflow] macOS something like that).

**Comment by webknjaz** [comment]
@hugovk could you go through all of the reusable workflows on the https://github.com/python/cpython/actions page, click on each ending up on a dedicated page (like https://github.com/python/cpython/actions/workflows/reusable-macos.yml), then use the three-dots top right menu to disable them? This will not affect their reusability, the calling workflows will still be able to invoke them. But the UI will show them as disabled. Here's an example of a project set up like this: https://github.com/ansible/awx-plugins/actions. The last one in the left sidebar is marked as `Disabled`.

File: .github/workflows/reusable-macos.yml (hunk 1 above)
Comment (webknjaz) [inline .github/workflows/reusable-macos.yml:1]:
```suggestion
name: ❌ [DO NOT CLICK] Reusable macOS
```

Comment (webknjaz) [inline .github/workflows/reusable-macos.yml]:
```suggestion
name: ❌ [DO NOT CLICK] Reusable macOS
```

File: .github/workflows/reusable-tsan.yml [hunk 2]
```diff
@@ -1,3 +1,5 @@
+name: Thread Sanitizer
```
Comment (webknjaz) [inline .github/workflows/reusable-tsan.yml]:
```suggestion
name: ❌ [DO NOT CLICK] Reusable Thread Sanitizer
```

File: .github/workflows/reusable-ubuntu.yml [hunk 3]
```diff
@@ -1,3 +1,5 @@
+name: Ubuntu
```
Comment (webknjaz) [inline .github/workflows/reusable-ubuntu.yml]:
```suggestion
name: ❌ [DO NOT CLICK] Reusable Ubuntu
```

File: .github/workflows/reusable-wasi.yml [hunk 4]
```diff
@@ -1,3 +1,5 @@
+name: WASI
```
Comment (webknjaz) [inline .github/workflows/reusable-wasi.yml]:
```suggestion
name: ❌ [DO NOT CLICK] Reusable WASI
```

File: .github/workflows/reusable-windows.yml [hunk 5]
```diff
@@ -1,3 +1,5 @@
+name: Windows
```
Comment (webknjaz) [inline .github/workflows/reusable-windows.yml]:
```suggestion
name: ❌ [DO NOT CLICK] Reusable Windows
```

**Review by webknjaz — COMMENTED** [review]
I was solving a similar concern while setting up new projects recently. I ended up putting a bit red cross with a “do not click” message in mine. I also marked the reusable workflows as disabled, so they are always at the bottom of the list.

Here's a demo of what it looks like in practice: https://github.com/ansible/awx-plugins/actions.

@hugovk would you like something similar here? I've copied my approach to the suggested changes but perhaps you'd want something shorter. LMK.

**Comment by webknjaz** [comment]
> define somekinds special naming that people notice that it's not a actual workflow.

If you go this way, and I think you should, I suggest also applying this to other `reusable-*.yml` workflow files. I realized that I've already added names to some of those, but it'd be nice to make them consistent in the UI, just like the filename prefix convention I introduced in the past.

**Comment by webknjaz** [comment]
Interestingly, I noticed 2 different workflows marked as `Docs` in the sidebar on the `Actions` tab. I was surprised, since I remember deleting one when introducing the reusable approach. So I clicked on the one that's not reusable and realized that it's coming from the `3.10` branch :exploding_head:

This is to say that marking the workflows as reusable explicitly in their names will be helpful.

**Comment by AA-Turner** [comment]
The workflows are now marked "disabled"; closing/reopening this PR to check that they are all still invoked.

**Comment by AA-Turner** [comment]
> perhaps you'd want something shorter

Let's avoid emoji, and perhaps just using "Reusable" will signal that they're special -- the "DO NOT CLICK" is a little distracting personally.

A

**Comment by webknjaz** [comment]
> The workflows are now marked "disabled"

@AA-Turner FYI, you missed a few like https://github.com/python/cpython/actions/workflows/reusable-docs.yml which was less obvious since it actually has a name (so it'll also need to be renamed following the same pattern).

File: .github/workflows/reusable-macos.yml (hunk 1 above)
Comment (webknjaz) [reply .github/workflows/reusable-macos.yml]: Variant 2
```suggestion
name: Reusable macOS
```

File: .github/workflows/reusable-tsan.yml (hunk 2 above)
Comment (webknjaz) [reply .github/workflows/reusable-tsan.yml]: Variant 2
```suggestion
name: Reusable Thread Sanitizer
```

File: .github/workflows/reusable-ubuntu.yml (hunk 3 above)
Comment (webknjaz) [reply .github/workflows/reusable-ubuntu.yml]: Variant 2
```suggestion
name: Reusable Ubuntu
```

File: .github/workflows/reusable-wasi.yml (hunk 4 above)
Comment (webknjaz) [reply .github/workflows/reusable-wasi.yml]: Variant 2
```suggestion
name: Reusable WASI
```

File: .github/workflows/reusable-windows.yml (hunk 5 above)
Comment (webknjaz) [reply .github/workflows/reusable-windows.yml]: Variant 2
```suggestion
name: Reusable Windows
```

**Review by webknjaz — COMMENTED** [review]
> > perhaps you'd want something shorter
> 
> Let's avoid emoji, and perhaps just using "Reusable" will signal that they're special -- the "DO NOT CLICK" is a little distracting personally (albeit explicit).

I suspected that people would not want emojis. The reason I started using them is that they are meant to be visually discouraging.

@hugovk @AA-Turner let me know what you want exactly, and I can update the suggested changes accordingly. Just `Reusable {{ X }}` or anything else?

**Comment by AA-Turner** [comment]
> let me know what you want exactly, and I can update the suggested changes accordingly. Just `Reusable {{ X }}` or anything else?

Either `Reusable {{ X }}` or `Reusable: {{ X }}` are fine by me.

A

**Comment by webknjaz** [comment]
Then just accept my variant 2 suggestions and wait for @corona10 to additionally make these changes to 3 more files, I suppose.

**Comment by corona10** [comment]
Hmm if there is no more objection,let's merge this PR?

**Review by AA-Turner — APPROVED** [review]

**Review by webknjaz — APPROVED** [review]

**Comment by webknjaz** [comment]
@AA-Turner perhaps also add backport request labels?

**Comment by hugovk** [comment]
Thanks all!

**Pull request merged**